RealmSwift/ThreadSafeReference.swift

```diff
@@ -122,6 +122,78 @@ public protocol ThreadConfined {
     }
 }
 
+/**
+ Object types conforming to `ThreadSafeWrappable` are able to have the `@ThreadSafe` property wrapper applied to them.
+
+ Note that only types defined by Realm can meaningfully conform to this protocol, and defining new
+ classes which attempt to conform to it will not make them work with `@ThreadSafe`
+ */
+public protocol ThreadSafeWrappable: ThreadConfined {}
```

Comment:
https://github.com/realm/realm-core/pull/4857 fixes the problem with using @ThreadSafe on Results and LinkingObjects, so this protocol should no longer be needed